cmd/auth.go

```diff
@@ -3,35 +3,34 @@ package cmd
 import (
 	"errors"
 	"fmt"
-	"net/url"
 	"os"
 	"reflect"
 	"strings"
 	"syscall"
 
-	"github.com/deis/workflow-cli/controller/client"
-	"github.com/deis/workflow-cli/controller/models/auth"
+	client "github.com/deis/controller-sdk-go"
+	"github.com/deis/controller-sdk-go/auth"
+	"github.com/deis/workflow-cli/settings"
 	"golang.org/x/crypto/ssh/terminal"
 )
 
 // Register creates a account on a Deis controller.
 func Register(controller string, username string, password string, email string,
 	sslVerify bool) error {
 
-	u, err := url.Parse(controller)
-	httpClient := client.CreateHTTPClient(sslVerify)
+	c, err := client.New(sslVerify, controller, "", "")
 
 	if err != nil {
 		return err
 	}
 
-	controllerURL, err := chooseScheme(*u)
+	tempClient, err := settings.Load()
 
-	if err != nil {
-		return err
+	if err == nil && tempClient.ControllerURL.Host == c.ControllerURL.Host {
+		c.Token = tempClient.Token
 	}
```

Comment:
This checks if the `*client.Client` properly loaded before trying to access `tempClient`, which otherwise could be `nil`.